service/src/test/java/tds/config/repositories/impl/ConfigRepositoryImplIntegrationTests.java

```diff
@@ -0,0 +1,158 @@
+package tds.config.repositories.impl;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import tds.config.ClientSystemFlag;
+import tds.config.ClientTestProperty;
+import tds.config.ConfigServiceApplication;
+import tds.config.TimeLimits;
+import tds.config.repositories.ConfigRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = ConfigServiceApplication.class)
+public class ConfigRepositoryImplIntegrationTests {
+
+    @Autowired
+    private ConfigRepository configRepository;
+
+    // -----------------------------------------------------------------------------------------------------------------
+    // ClientSystemFlag Tests
+    // -----------------------------------------------------------------------------------------------------------------
+    @Test
+    public void shouldGetAClientSystemFlagsList() {
+        final String clientName = "SBAC_PT";
+
+        List<ClientSystemFlag> result = configRepository.getClientSystemFlags(clientName);
+
+        assertThat(result).isNotNull();
+        assertThat(result.size()).isEqualTo(16);
+        result.forEach(r -> {
+            assertThat(r.getAuditObject()).isNotNull();
+            assertThat(r.getClientName()).isEqualTo(clientName);
+        });
+    }
+
+    @Test
+    public void shouldGetOptionalEmptyClientSystemFlagsForAnInvalidClientName() {
+        final String clientName = "foo";
+
+        List<ClientSystemFlag> result = configRepository.getClientSystemFlags(clientName);
+
+        assertThat(result.size()).isEqualTo(0);
+    }
+
+    // -----------------------------------------------------------------------------------------------------------------
+    // ClientTestProperty Tests
+    // -----------------------------------------------------------------------------------------------------------------
+    @Test
+    public void shouldGetAClientTestProperty() throws Exception {
+        final String clientName = "SBAC_PT";
+        final String assessmentId = "SBAC Math 3-MATH-3";
+
+        Optional<ClientTestProperty> result = configRepository.getClientTestProperty(clientName, assessmentId);
+
+        assertThat(result).isPresent();
+        assertThat(result.get().getClientName()).isEqualTo("SBAC_PT");
+        assertThat(result.get().getAssessmentId()).isEqualTo("SBAC Math 3-MATH-3");
+        assertThat(result.get().getMaxOpportunities()).isEqualTo(3);
+        assertThat(result.get().getPrefetch()).isEqualTo(2);
+        assertThat(result.get().getIsPrintable()).isEqualTo(false);
+        assertThat(result.get().getIsSelectable()).isEqualTo(true);
```

Comment:
Don't think we've discussed it, but I personally like `assertThat(...).isTrue()` instead of `.isEqualTo(true)`.  I feel it's easier to read.